File: benchmark/python-app/images.py
import asyncio
import uuid
from datetime import datetime


class Image:
    def __init__(self):
        self.uuid = self._generate_uuid()
        self.last_modified = self._get_time()

    def _generate_uuid(self):
        return str(uuid.uuid4())

    def _get_time(self):
        return datetime.now()


async def download():
    await asyncio.sleep(0.2)


async def save(image: Image):
    await asyncio.sleep(0.5)
    del image

File: benchmark/go-app/images.go
package main

import (
	"time"

	"github.com/google/uuid"
)

type Image struct {
	ImageUUID string
	LastModified time.Time
}

func NewImage() *Image {
	id := uuid.New().String()
	lastModified := time.Now()

	return &Image{
		ImageUUID:    id,
		LastModified: lastModified,
	}
}

func download(key string) error {
	time.Sleep(time.Millisecond * 5)
	return nil
}

func save(c *Image) error {
	time.Sleep(time.Millisecond * 2)
	return nil
}
